File: app/src/main/java/zjm/cst/dhu/readattitude/mvp/ui/adapter/MainFragmentPagerAdapter.java
package zjm.cst.dhu.readattitude.mvp.ui.adapter;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by zjm on 2017/5/9.
 */

public class MainFragmentPagerAdapter extends FragmentPagerAdapter {

    private List<Fragment> mFragmentList = new ArrayList<>();
    private List<String> mTitles;

    public MainFragmentPagerAdapter(FragmentManager fm, List<String> titles, List<Fragment> fragmentList) {
        super(fm);
        mTitles = titles;
        mFragmentList = fragmentList;
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return mTitles.get(position);
    }

    @Override
    public Fragment getItem(int position) {
        return mFragmentList.get(position);
    }

    @Override
    public int getCount() {
        return mFragmentList.size();
    }
}

File: app/src/main/kotlin/zjm/cst/dhu/readattitude/mvp/ui/adapter/MainFragmentPagerAdapter.kt
package zjm.cst.dhu.readattitude.mvp.ui.adapter

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter

/**
 * Created by zjm on 2017/5/9.
 */

class MainFragmentPagerAdapter(fm: FragmentManager,
                               private val mTitles: List<String>,
                               private val mFragmentList: List<Fragment>)
    : FragmentPagerAdapter(fm) {


    override fun getPageTitle(position: Int):
            CharSequence {
        return mTitles[position]
    }

    override fun getItem(position: Int):
            Fragment {
        return mFragmentList[position]
    }

    override fun getCount():
            Int {
        return mFragmentList.size
    }
}
